### ISSUE TITLE
On the TorchTitan Infrastructure Build-out (VLM)

### ISSUE THREAD

**Issue by Joluck** [pytorch/torchtitan#2008]
In the past, I’ve always trained models with the Lightning framework; now I’d like to switch to a more efficient one (TorchTitan or Megatron). However, I’ve run into a few questions and would appreciate your advice:
Can I simply import the encoder part straight from Hugging Face Transformers? (In VLM, the encoder usually accounts for only a small fraction of the parameters, so in my view it doesn’t need tensor-parallelism, etc.)

**Comment by tianyu-l** [comment]
That's doable. See the flux training for an example of how to use HF encoders with torchtitan.

https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/flux/train.py#L58-L65

In this example, we did apply FSDP to the HF encoders, but you don't have to, assuming you'd freeze the encoders and not train them. (If you train them, then gradients need to be reduced across all DP ranks even if they are not sharded.)

cc @lkhphuc

**Comment by Joluck** [comment]
> That's doable. See the flux training for an example of how to use HF encoders with torchtitan.
> 
> https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/flux/train.py#L58-L65
> 
> In this example, we did apply FSDP to the HF encoders, but you don't have to, assuming you'd freeze the encoders and not train them. (If you train them, then gradients need to be reduced across all DP ranks even if they are not sharded.)
> 
> cc [@lkhphuc](https://github.com/lkhphuc)


Thank you very much for your reply. I will try it step by step. So, at the moment, if I want to train the encoder part, I must write it in the FSDP format, right?

**Comment by tianyu-l** [comment]
If you want to train it, but keep it replicated instead of sharded, you could consider the alternative API `replicate`, which will do the gradient all-reduce for you.

Examples can be found in the WIP PR is https://github.com/pytorch/torchtitan/pull/1714

cc @anshul-si

**Comment by Joluck** [comment]
> If you want to train it, but keep it replicated instead of sharded, you could consider the alternative API `replicate`, which will do the gradient all-reduce for you.
> 
> Examples can be found in the WIP PR is [#1714](https://github.com/pytorch/torchtitan/pull/1714)
> 
> cc [@anshul-si](https://github.com/anshul-si)

Thank you so much! I’m currently training **Qwen3-0.6B** on **4×4090 GPUs** to compare the **memory usage and speed** between **TorchTitan** and **Lightning + DeepSpeed**.
 Is there any **official benchmark comparison** available?
